traffic_ops/traffic_ops_golang/cdn/capacity.go

```diff
@@ -0,0 +1,325 @@
+	profileThresholds := map[string]float64{}
+	for rows.Next() {
+		profile := ""
+		threshStr := ""
+		if err := rows.Scan(&profile, &threshStr); err != nil {
+			return nil, errors.New("scanning thresholds: " + err.Error())
+		}
+		threshStr = strings.TrimPrefix(threshStr, ">")
+		thresh, err := strconv.ParseFloat(threshStr, 64)
+		if err != nil {
+			return nil, errors.New("profile '" + profile + "' health.threshold.availableBandwidthInKbps is not a number")
+		}
+		profileThresholds[profile] = thresh
+	}
+	return profileThresholds, nil
+}
+
+func getCRStates(monitorFQDN string, client *http.Client) (CRStates, error) {
+	path := `/publish/CrStates`
+	resp, err := client.Get("http://" + monitorFQDN + path)
+	if err != nil {
+		return CRStates{}, errors.New("getting CRStates from Monitor '" + monitorFQDN + "': " + err.Error())
+	}
+	defer resp.Body.Close()
+
+	crs := CRStates{}
+	if err := json.NewDecoder(resp.Body).Decode(&crs); err != nil {
+		return CRStates{}, errors.New("decoding CRStates from monitor '" + monitorFQDN + "': " + err.Error())
+	}
+	return crs, nil
+}
+
+func getCRConfig(monitorFQDN string, client *http.Client) (CRConfig, error) {
+	path := `/publish/CrConfig`
+	resp, err := client.Get("http://" + monitorFQDN + path)
+	if err != nil {
+		return CRConfig{}, errors.New("getting CRConfig from Monitor '" + monitorFQDN + "': " + err.Error())
+	}
+	defer resp.Body.Close()
+	crs := CRConfig{}
+	if err := json.NewDecoder(resp.Body).Decode(&crs); err != nil {
+		return CRConfig{}, errors.New("decoding CRConfig from monitor '" + monitorFQDN + "': " + err.Error())
+	}
+	return crs, nil
+}
+
+// CacheStats contains the Monitor CacheStats needed by Cachedata. It is NOT the full object served by the Monitor, but only the data required by the caches stats endpoint.
+type CacheStats struct {
+	Caches map[tc.CacheName]CacheStat `json:"caches"`
+}
+
+type CacheStat struct {
+	KBPS    []CacheStatData `json:"kbps"`
+	MaxKBPS []CacheStatData `json:"maxKbps"`
+}
+
+type CacheStatData struct {
+	Value float64 `json:"value,string"`
+}
+
+// getCacheStats gets the cache stats from the given monitor. It takes stats, a slice of stat names; and cacheStats, an object to deserialize stats into. The cacheStats type must be of the form struct {caches map[tc.CacheName]struct{statName []struct{value float64}}} with the desired stats, with appropriate member names or tags.
+func getCacheStats(monitorFQDN string, client *http.Client, stats []string, cacheStats interface{}) error {
+	path := `/publish/CacheStats`
+	if len(stats) > 0 {
+		path += `?stats=` + strings.Join(stats, `,`)
+	}
+	resp, err := client.Get("http://" + monitorFQDN + path)
+	if err != nil {
+		return errors.New("getting CacheStats from Monitor '" + monitorFQDN + "': " + err.Error())
+	}
+	defer resp.Body.Close()
+	if err := json.NewDecoder(resp.Body).Decode(cacheStats); err != nil {
+		return errors.New("decoding CacheStats from monitor '" + monitorFQDN + "': " + err.Error())
```

Comment:
i changed it to `/CacheStats` and now i'm getting this error when hitting an online monitor:

`ERROR: api.go:122: 2019-08-16T15:14:35.164661-06:00: 127.0.0.1:65348 getting cache stats for CDN 'cdn-1' monitor 'cdn-trafficmonitor-foo.bar.net:80': decoding CacheStats from monitor 'cdn-trafficmonitor-foo.bar.net:80': invalid character '<' looking for beginning of value
`

Reply:
nevermind. publish is the json version. duh.